packages/core/src/Provider/types/StakePool.ts

```diff
@@ -0,0 +1,136 @@
+import { Address, PoolId, Hash16 } from '@cardano-ogmios/schema';
+import { ExtendedStakePoolMetadata } from './ExtendedStakePoolMetadata';
+import { Lovelace } from '../../Ogmios';
+
+/**
+ * Within range [0; 1]
+ */
+export type Percent = number;
+
+export type TransactionId = Hash16;
+
+/**
+ * https://github.com/cardano-foundation/CIPs/blob/master/CIP-0006/CIP-0006.md#on-chain-referenced-main-metadata-file
+ */
+export interface StakePoolMetadata {
+  /**
+   * Pool ticker. uppercase
+   * 5 Characters Maximum, Uppercase letters and numbers
+   */
+  ticker: string;
+  /**
+   * A name for the pool
+   * 50 Characters Maximum
+   */
+  name: string;
+  /**
+   * Text that describes the pool
+   * 50 Characters Maximum
+   */
+  description: string;
+  /**
+   * A website URL for the pool
+   * 64 Characters Maximum, must be a valid URL
+   */
+  homepage: string;
+  /**
+   * A URL for extended metadata
+   * 128 Characters Maximum, must be a valid URL
+   */
+  extDataUrl?: string;
+  /**
+   * A URL with the extended metadata signature
+   * 128 Characters Maximum, must be a valid URL
+   */
+  extSigUrl?: string;
+  /**
+   * the public Key for verification
+   * optional, 68 Characters
+   */
+  extVkey?: Hash16; // Review: is this the correct type alias?
+  /**
+   * Extended metadata defined by CIP-6
+   */
+  ext?: ExtendedStakePoolMetadata;
+}
+
+export interface StakePool {
```

Comment:
We should model the on-chain aspect as closely to the ledger model as possible, i.e extend the Ogmios `PoolParameters` interface.